### PR TITLE
Arm/refactor/task breakdown

### PR DESCRIPTION
**Pull request by arminsabouri**
And fixed the current liting and fmt warnings we we're getting. Hopefully CI passes now

### CODE REVIEW

**Comment by arminsabouri** [comment]
Will open up to review once local testing passes

File: crates/consensus/authority/src/block_fetcher.rs [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+                    TryRecvError::Disconnected => {
+                        error!(target: "consensus::authority", "Block import channel disconnected");
+                        continue;
+                    }
+                },
+            };
+
+            let block = new_block.block.block.clone();
+            info!(target: "consensus::authority", ?block, "Recieved new block from peer");
+
```
Comment (scottmillner) [inline crates/consensus/authority/src/block_fetcher.rs:90]: why was this commented out?

File: crates/consensus/utils.rs [hunk 2]
```diff
@@ -0,0 +1,5 @@
+
```
Comment (scottmillner) [inline crates/consensus/utils.rs]: should this file be deleted?

File: crates/consensus/authority/src/block_fetcher.rs (hunk 1 above)
Comment (arminsabouri) [reply crates/consensus/authority/src/block_fetcher.rs:90]: I was deciding where to put these checks. PoA consenus checks we're kinda spread around all over the place. So I consolidated them all in the latest commit 
[1911a1c](https://github.com/botanix-labs/Macbeth/pull/78/commits/1911a1c08a34f7b20ced52984d1bc56be5e7aeca)

File: crates/consensus/utils.rs (hunk 2 above)
Comment (arminsabouri) [reply crates/consensus/utils.rs]: Oof yes sorry

**Comment by arminsabouri** [comment]
Per ussual the refactor PR blew up in scope and I ended up doing more than I intended. Let's hop on a call to discuss this PR. Should help the review process

File: bin/reth/src/poa/mod.rs [hunk 3]
```diff
@@ -549,6 +555,14 @@ impl<Ext: RethCliExt> PoaNodeCommand<Ext> {
             block_production_task.start_task().await;
         }));
 
+        ctx.task_executor.spawn(Box::pin(async move {
```
Comment (arminsabouri) [inline bin/reth/src/poa/mod.rs]: Should these critical?

**Review by scottmillner — APPROVED** [review]
🔥 🔥 🔥

**Review by BatiGencho — APPROVED** [review]

File: bin/reth/src/poa/mod.rs (hunk 3 above)
Comment (scottmillner) [reply bin/reth/src/poa/mod.rs]: seems like they should be right? if any of those tasks die, any reason we'd want to continue running the node?

**Review by scottmillner — APPROVED** [review]

File: bin/reth/src/poa/mod.rs (hunk 3 above)
Comment (arminsabouri) [reply bin/reth/src/poa/mod.rs]: Fixed

**Pull request merged**